containers/payments/PaymentVerificationModal.js

```diff
@@ -1,26 +1,64 @@
-import React from 'react';
+import React, { useState, useRef } from 'react';
 import PropTypes from 'prop-types';
-import { FormModal, Alert, Loader, useNotifications, useLoading, useApi, useConfig } from 'react-components';
+import {
+    FormModal,
+    Alert,
+    Loader,
+    Button,
+    ResetButton,
+    useNotifications,
+    useApi,
+    useConfig,
+    PrimaryButton
+} from 'react-components';
 import { c } from 'ttag';
 import tabSvg from 'design-system/assets/img/pm-images/tab.svg';
+import errorSvg from 'design-system/assets/img/pm-images/error.svg';
 
 import { toParams, process } from './paymentTokenHelper';
 
+const STEPS = {
+    REDIRECT: 'redirect',
+    REDIRECTING: 'redirecting',
+    REDIRECTED: 'redirected',
+    FAIL: 'fail'
+};
+
+const PROCESSING_DELAY = 5000;
+
 const PaymentVerificationModal = ({ params, token, approvalURL, onSubmit, ...rest }) => {
-    const [loading, withLoading] = useLoading();
+    const TITLES = {
+        redirect: c('Title').t`Card verification`,
+        redirecting: c('Title').t`Processing...`,
+        redirected: c('Title').t`Payment verification in progress`,
+        fail: c('Title').t`3-D secure payment verification failed`
+    };
+    const [step, setStep] = useState(STEPS.REDIRECT);
+    const [error, setError] = useState();
     const api = useApi();
     const { createNotification } = useNotifications();
-    const title = loading ? c('Title').t`Payment verification in progress` : c('Title').t`Payment verification`;
     const { SECURE_URL: secureURL } = useConfig();
+    const abortRef = useRef();
+
+    const handleCancel = () => {
+        abortRef.current && abortRef.current.abort();
+        rest.onClose();
+    };
 
     const handleSubmit = async () => {
         try {
-            await process({ Token: token, api, approvalURL, secureURL });
+            setStep(STEPS.REDIRECTING);
+            setTimeout(() => {
+                setStep(STEPS.REDIRECTED);
+            }, PROCESSING_DELAY);
```

Comment:
Why do we wait 5 seconds instead of for some successful response?

Reply:
The problem is if there is no successful response (eg. pop-up blocker).